src/Microsoft.Sbom.Common/Utils/CommonSPDXUtils.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Sbom.Common.Utils;
+
+/// <summary>
+/// Provides utility methods for SPDX objects.
+/// </summary>
+public class CommonSPDXUtils
+{
+    /// <summary>
+    /// Only these chars are allowed in a SPDX id. Replace all other chars with '-'.
+    /// </summary>
+    private static readonly Regex SpdxIdAllowedCharsRegex = new Regex("[^a-zA-Z0-9.-]");
+
+    /// <summary>
+    /// Returns the SPDX-compliant package ID.
+    /// </summary>
+    public string GenerateSpdxPackageId(string id) => $"{Constants.SPDXRefPackage}-{GetStringHash(id)}";
+
+    /// <summary>
+    /// Returns the SPDX-compliant file ID.
+    /// </summary>
+    public string GenerateSpdxFileId(string fileName, string sha1Value)
+    {
+        var spdxFileId = $"{Constants.SPDXRefFile}-{fileName}-{sha1Value}";
+        return SpdxIdAllowedCharsRegex.Replace(spdxFileId, "-");
+    }
+
+    /// <summary>
+    /// Returns the SPDX-compliant external document ID.
+    /// </summary>
+    public string GenerateSpdxExternalDocumentId(string fileName, string sha1Value)
+    {
+        var spdxExternalDocumentId = $"{Constants.SPDXRefExternalDocument}-{fileName}-{sha1Value}";
+        return SpdxIdAllowedCharsRegex.Replace(spdxExternalDocumentId, "-");
+    }
+
+    /// <summary>
+    /// Compute the SHA256 string representation (omitting dashes) of a given string
+    /// </summary>
+    /// <remarks>
+    /// TODO:  refactor this into Core as similar functionality is duplicated in a few different places in the codebase
```

Comment:
Does this PR complete the TODO ?

Reply:
This comment was from someone else. I assume that by "Core" they meant Common since we don't have a Core project, so I think this TODO is resolved with this refactor. If you agree as well, I can remove it.